src/mlpack/methods/ann/ffn.hpp

```diff
@@ -72,56 +75,73 @@ class FFN
       InitializationRuleType initializeRule = InitializationRuleType());
 
   //! Copy constructor.
-  FFN(const FFN&);
-
+  FFN(const FFN& other);
   //! Move constructor.
-  FFN(FFN&&);
-
-  //! Copy/move assignment operator.
-  FFN& operator = (FFN);
+  FFN(FFN&& other);
+  //! Copy operator.
+  FFN& operator=(const FFN& other);
+  //! Move assignment operator.
+  FFN& operator=(FFN&& other);
 
-  //! Destructor to release allocated memory.
-  ~FFN();
+  /**
+   * Add a new layer to the model.
+   *
+   * @param args The layer parameter.
+   */
+  template <class LayerType, class... Args>
```

Comment:
Small doubt: Why do we use `class` here instead of `typename`? Aren't they semantically the same?

Reply:
Yeah, they are equivalent.  I guess the rest of the code uses `typename`... I changed it in 1645bb22.